source/MaterialXRuntime/Codegen/RtCodegenImpl.h

```diff
@@ -0,0 +1,45 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved. See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTCODEGENIMPL_H
+#define MATERIALX_RTCODEGENIMPL_H
+
+/// @file RtCodegenImpl.h
+/// TODO: Docs
+
+#include <MaterialXRuntime/RtNodeImpl.h>
+#include <MaterialXRuntime/RtNode.h>
+
+#include <MaterialXGenShader/GenContext.h>
+
+namespace MaterialX
+{
+
+/// @class RtCodegenImpl
+/// Abstract base class for all node implementations that use shader code generation.
```

Comment:
Isn't this not an abstract class as you have no pure virtual functions?

Reply:
As a runtime prim it's abstract in the sense that you cannot create instances of this prim. But you are right that in a C++ sense it's not abstract since it has no pure virtual methods. I will update the comment.